packages/ampd-sdk/src/config.rs

```diff
@@ -0,0 +1,202 @@
+    pub fn add_env_source(self, prefix: &str) -> Self {
+        Self(self.0.add_source(config::Environment::with_prefix(prefix)))
+    }
+
+    /// Builds the config from the sources.
+    ///
+    /// The config is deserialized from the sources into a `Config` struct.
+    pub fn build(self) -> Result<Config, Error> {
+        self.0
+            .build()
+            .and_then(|config| config.try_deserialize::<Config>())
+            .change_context(Error::Build)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::fs;
+    use std::str::FromStr;
+
+    use axelar_wasm_std::assert_err_contains;
+    use tempfile::TempDir;
+
+    use super::*;
+
+    #[test]
+    fn config_loads_from_default_sources() {
+        let ampd_url = "http://localhost:8080";
+        let chain_name = "some-chain";
+
+        temp_env::with_vars(
+            vec![
+                (format!("{DEFAULT_CONFIG_PREFIX}_AMPD_URL"), Some(ampd_url)),
+                (
+                    format!("{DEFAULT_CONFIG_PREFIX}_CHAIN_NAME"),
+                    Some(chain_name),
+                ),
+            ],
+            || {
+                let config = Config::from_default_sources().unwrap();
+
+                assert_eq!(config.ampd_url, Url::new_sensitive(ampd_url).unwrap());
+                assert_eq!(config.chain_name, ChainName::from_str(chain_name).unwrap());
+            },
+        );
+    }
+
+    #[test]
+    fn config_loads_from_custom_env_vars() {
+        let ampd_url = "http://localhost:8080";
+        let chain_name = "some-chain";
+        let prefix = "MY_HANDLER";
+
+        temp_env::with_vars(
+            vec![
+                (format!("{prefix}_AMPD_URL"), Some(ampd_url)),
+                (format!("{prefix}_CHAIN_NAME"), Some(chain_name)),
+            ],
+            || {
+                let config = Config::builder().add_env_source(prefix).build().unwrap();
+
+                assert_eq!(config.ampd_url, Url::new_sensitive(ampd_url).unwrap());
+                assert_eq!(config.chain_name, ChainName::from_str(chain_name).unwrap());
+            },
+        );
+    }
+
+    #[test]
+    fn config_loads_from_custom_file() {
+        let ampd_url = "http://localhost:8080";
+        let chain_name = "some-chain";
+
+        let temp_dir = TempDir::new().unwrap();
+        let config_path = temp_dir.path().join("custom-config.toml");
+        let content = format!(
+            r#"
+            ampd_url="{ampd_url}"
+            chain_name="{chain_name}"
+            "#
+        );
+        fs::write(&config_path, content).unwrap();
+
+        let config = Config::builder()
+            .add_file_source(config_path.to_str().unwrap(), true)
+            .build()
+            .unwrap();
+
+        assert_eq!(config.ampd_url, Url::new_sensitive(ampd_url).unwrap());
+        assert_eq!(config.chain_name, ChainName::from_str(chain_name).unwrap());
+    }
+
+    #[test]
+    fn config_load_fails_if_field_is_missing() {
```

Comment:
why don't we want to fall back on defaults?

Reply:
I didn't implement `Default` for the config, I was going to, but I wasn't sure if it was a good idea to have a default `chain_name`. What do you think?

Reply:
oh the chain name is in there, yeah we don't have a default for that. Can we make some fields required and others fall back to a default?